docs/semgrep-ci/configuration-reference.md

```diff
@@ -22,124 +22,243 @@ Object.entries(frontMatter).filter(
-- If you do **not** use Semgrep App with Semgrep in CI (stand-alone setup), blocking findings encompass all Semgrep findings. Any finding in this setup blocks your PRs or MRs.
+Set `SEMGREP_ENABLE_VERSION_CHECK` to 0 to **disable** version checks when running `semgrep ci`. By default, Semgrep checks for new versions.
 
-By default, Semgrep does not block your pipeline when it encounters an internal error. Semgrep suppresses all errors and does not surface them to the CI provider. In case of an internal error, Semgrep sends an anonymous crash report to a crash-reporting server and does not block your CI pipeline. To change the default configuration, see the sections below.
+Example:
 
-### Configuration options for blocking findings and errors
+```bash
+# Disable version checks when running semgrep ci:
+export SEMGREP_ENABLE_VERSION_CHECK="0"
+```
 
-Configure, change or revert to the default setup of blocking findings and errors in your CI pipeline using the following options in CI configuration file:
+### `SEMGREP_GHA_MIN_FETCH_DEPTH`
 
-| CI option                                      | Description                         |
-|------------------------------------------------|-------------------------------------|
-| `semgrep ci` or `semgrep ci --suppress-errors` | Default: CI **fails** on blocking findings, CI **passes** on internal errors.  |
-| `semgrep ci --no-suppress-errors`              | CI **fails** on blocking findings, CI **fails** on internal errors.            |
-| <code>semgrep ci &vert;&vert; true</code>      | CI **passes** on blocking findings, CI **passes** on internal errors.          |
- 
-To change this configuration, insert one of the configuration options (flags) after the following keys in in CI YAML configuration file of Semgrep:
-- On GitHub, insert the flag after the `run` key (for example, `run: semgrep ci --suppress-errors` to state the default option).
-- On GitLab, insert the flag after the `script` key (for example, `script: semgrep ci --suppress-errors` to state the default option).
-- Insert these flags in an equivalent key in configuration files of other CI providers.
- 
-See the [Examples of blocking findings and errors configuration](#examples-of-blocking-findings-and-errors-configuration) below.
- 
-:::info
-- For more information about specific Semgrep exit codes, see [CLI reference](../../cli-reference/#exit-codes).
-- This functionality replaces the audit mode `SEMGREP_AUDIT_ON` (collecting findings silently for [Semgrep App > Findings](https://semgrep.dev/manage/findings)).
-:::
+Set `SEMGREP_GHA_MIN_FETCH_DEPTH` to configure the **minimum** number of commits `semgrep ci` fetches from `remote` when calculating the merge-base in GitHub Actions. Having more commits available helps Semgrep determine what changes came from the current pull request, fixing issues where Semgrep would otherwise report findings that were not touched in a given pull request. This value is set to 0 by default. 
```

Comment:
The following sentence lacks a call to action for a user (had to read it a couple of times to figure out one): `Having more commits available helps Semgrep determine what changes came from the current pull request, fixing issues where Semgrep would otherwise report findings that were not touched in a given pull request. This value is set to 0 by default.`

Maybe something like:

```
For optimal performance, use `SEMGREP_GHA_MIN_FETCH_DEPTH` with more commits. Having more commits available helps Semgrep determine what changes came from the current pull request, fixing issues where Semgrep would otherwise report findings that were not touched in a given pull request. This value is set to 0 by default.
```

Note: `More commits` sounds a bit vague also. Do we mean 3 commits or 20?

Reply:
Some context for this - it was made to deal with diff-aware findings duplication for a particular use case. We can't say for sure "how many" commits back the user needs. What i have done is to add a note saying "don't set this variable unless you are encountering findings duplication in your diff aware scans".

Reply:
@s-santillan I like the added note! ("don't set this variable unless you are encountering findings duplication in your diff aware scans")